### PR TITLE
Laying ground work to UI development

### PR DESCRIPTION
**Pull request by justinbarry**
Note: Augur-node v2 transition is still in progress.

* docker:build:all working
* Deploy augur-artifacts version of contracts.json
* Remove broke files
* Restore contracts.json
* Add gas-limit dp helper
* Fixes to get augur-ui to build
* Add explicit monorepo deps
* Update build scripts
* Remove unused file
* Remove unneeded .nvmrc files
* Move rep-faucet
* Declare augur-tool's dependency on contract-deps-ethers
* Create a typed union for network names
* Upgrade TS version
* Migrate DP scripts to typescript
* Add topics to event log types
* Ignore pouchdb files
* Migrate speedomatic to monot repo and transition to TS.
* Augur node replace augur.js with augur-api
* Update mono-repo create script

ref #1281, #1352

### CODE REVIEW

**Review by brianosaurus — APPROVED** [review]
LGTM (it is just a merge after all) .... Stuff I did see will get cleaned up in my next PR.

File: packages/augur-api/src/ethereum/types.ts [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
 
 export interface ParsedLog {
-  blockNumber?: number;
-  blockHash?: string;
-  transactionIndex?: number;
-  removed?: boolean;
-  transactionLogIndex?: number;
-  transactionHash?: string;
-  logIndex?: number;
+  blockNumber: number;
```
Comment (adrake33) [inline packages/augur-api/src/ethereum/types.ts:19]: It seems odd that these were made optional. I suppose as long as it still runs as expected, it's good to change them.

Comment (justinbarry) [reply packages/augur-api/src/ethereum/types.ts:19]: I think they were made required on this branch.

Comment (adrake33) [reply packages/augur-api/src/ethereum/types.ts:19]: Right, I just mean I don't know why they were originally made optional. Seems like they should always be expected to be present.

Comment (justinbarry) [reply packages/augur-api/src/ethereum/types.ts:19]: Ah. Yeah, hopefully marking these required makes life simpler.

**Review by adrake33 — APPROVED** [review]
LGTM as well.

**Pull request merged**